consensus-engine/src/lib.rs

```diff
@@ -390,6 +390,10 @@ impl<O: Overlay> Carnot<O> {
             Err(e) => Err(e),
         }
     }
+
+    pub fn prune_older_blocks_by_view(&mut self, threshold_view: View) {
```

Comment:
nit: maybe a warning that we shouldn't prune blocks that are more recent than the last committed block (i.e. put a warning / assert that checks wether threshold_view < last_committed_block().view

Reply:
Yep, safer that way,, thanks!